usekase-coroutines/src/main/kotlin/guru/stefma/cleancomponents/usecase/coroutines/CoroutineUseCase.kt

```diff
@@ -1,17 +1,41 @@
 package guru.stefma.cleancomponents.usecase.coroutines
 
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
 /**
- * A UseCase is a "actor" or "transformer" in the clean architecture.
+ * A UseCase is an "actor" or "transformer" in the Clean Architecture.
  *
- * Use this interface as a default implementation of an interface which have [params][P] and a [result][R].
+ * Use this base class to create a use case with input [parameters][Params] and an expected [result][Type].
  *
- * Simply call [execute] with the given [params][P] to get the [result][R].
+ * By default, once your overridden [execute] method returns, the callback lambdas passed to [invoke] will execute
+ * on the [main thread][Dispatchers.Main]. This requires a runtime dependency of coroutines that adds a main
+ * dispatcher. You can override [callbackDispatcher] to change this behavior.
  *
- * @param P the params you want to put in to the UseCase
- * @param R the result value which will be emitted by this UseCase
+ * @param Type The type which you expect to be returned by the use case.
+ * @param Params The parameters which the use case needs as an input.
  */
-interface CoroutineUseCase<R, P> {
+abstract class CoroutineUseCase<out Type, in Params> where Type : Any {
+
+    open val callbackDispatcher: CoroutineContext = Dispatchers.Main
```

Comment:
As you have already written in your PR comment; this require a special coroutine dependency. Like `android` or swing...
Can you please write this into the readme as well. Thanks 👍

Reply:
Oh.. and btw.
Why don't we have this dependecy in the jvm sample? 🤔